manifests/backend.pp

```diff
@@ -42,6 +42,7 @@
   Stdlib::Port $url_port = 8080,
   String $username = 'admin',
   String $password = 'P@ssw0rd!',
+  Array $bcrypt_dependencies = ['make','gcc']
```

Comment:
It was sounding like the sensu-go code would just expose the bcrypt hash that's stored in etcd built into backend.  I don't think they can convert the stored bcrypt hash into sha or anything since the backend only stores the bcrypt hash which can't be decrypted as far as I am aware.

As for having gcc and make, maybe my style is not common but most systems I manage don't have ability to build software since everything comes from package manager.  The most popular modules in forge for make and gcc both use `ensure_packages` so our use of `ensure_packages` would not collide with those modules.  People can pass an empty array or we could add a `manage_bcrypt_dependencies` parameter too.

Reply:
With this patch you will have to have gcc on a system to use Sensu and that's asking too much as it is not allowed in many environments. We need to work with Sensu engineering to find another way that can be implemented with existing ruby libraries that do not depend on the system having a compiler.

Reply:
We could have sensuctl take a string and output the bcrypt info and then ask that the password in our manifest be that bcrypt'd string instead of a real password. This is similar to what people do with the user resource and setting passwords. This way we're just doing string comparison.

Reply:
I like the idea of changing `password` to `password_hash`.  We could provide a Puppet function that uses bcrypt that is completely optional and would allow a site to do something like this:

```puppet
sensu_user { 'admin':
  ensure               => 'present',
  password_hash => sensu_password('P@ssw0rd!')
}
```
This would mean the `sensu::backend` parameter would become `password_hash` and then we could document on possible way of declaring the class is like this:

```puppet
class { 'sensu::backend':
  password_hash => sensu_password('P@ssw0rd!')
}
```
I saw other sensu-go issues where it was being proposed that they would actually only expose `password_hash` key as input when defining users with JSON so this would work nicely with Puppet just expecting the password hash.

Reply:
@ghoneycutt The only issue with accepting a hash is that `sensuctl configure` expects a plain text password so we'd still need the plain text password.

Reply:
If it was a function that would be OK, since it would only be on the puppet masters

Reply:
Please comment in the existing issue we have

Reply:
For the function approach, we'd still need `password` parameter for `sensu::backend` and I suppose just document that the master needs to have bcrypt installed.   Would we want a helper class like `sensu::puppetserver` or something that does this:

```puppet
package { 'bcrypt':
  ensure   => 'installed',
  provider => 'puppetserver_gem',
}
```

This would require : https://github.com/puppetlabs/puppetlabs-puppetserver_gem.

Reply:
That's a good approach. Let's see how Sensu wants to handle this before you write any real code around it.